libbeat/common/transport/httpcommon/httpcommon.go

```diff
@@ -0,0 +1,338 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package httpcommon
+
+import (
+	"net/http"
+	"time"
+
+	"go.elastic.co/apm/module/apmhttp"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transport"
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+// HTTPTransportSettings provides common HTTP settings for HTTP clients.
+type HTTPTransportSettings struct {
```

Comment:
To me `HTTPTransportSettings` does not sound idiomatic. I would rather call it `HTTPTransport`. In Golang, `Transport` (and `DefaultTransport`) contains the same settings. Also, a Transport is also a `RoundTripper` just like your `HTTPTransportSettigns`. Normally, I do not care about naming that much, but as this is going into libbeat to be shared among Beats, I would like this to be either renamed or provide valid reason why this name fits the data structure.

Reply:
This is no `Transport` or `RoundTripper`. The httpcommon package provides common structs, that you can reuse/inline in your Config structs. This type is a `RoundTripper` builder, that can be initialized via `go-ufc`, but does not implement a RoundTripper itself. I named it HTTPTransportSettings, to make this intend more clear.

The `http.Transport` requires you to initialize by creating your struct, but it does not integrate with our configuration system. Once `http.Transport` is used the first time, the struct must not be used anymore. This is rather a common initialization pattern, but no builder.

Reply:
Ok, it is a bit confusing to have a `Transport` setting and a `Transport` in clients. But I get the big picture.